docs/core-concepts/components.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Components
+---
+
+This document provides an overview of the components required for a fully functional and operational Karmada setup.
+
+![components](../resources/general/components.png)
+
+## Control Plane Components
+
+A complete and working Karmada control plane consists of the following components. 
+The `karmada-agent` might be optional, that depends on 
+[Cluster Registration Mode](https://kubernetes.io/docs/concepts/overview/components/).
+
+### karmada-apiserver
+
+The API server is a component of the Karmada control plane that exposes the Karmada API in addition to the Kubernetes API. 
+The API server is the front end of the Karmada control plane.
+
+Karmada API server directly uses the implementation of `kube-apiserver` from Kubernetes, which is the reason why Karmada 
+is naturally compatible with Kubernetes API. That makes integration with the Kubernetes ecosystem very simple for Karmada, 
+such as allowing users to use `kubectl` to operate Karmada, 
+[integrating with ArgoCD](https://karmada.io/docs/next/userguide/cicd/working-with-argocd), 
+[integrating with Flux](https://karmada.io/docs/next/userguide/cicd/working-with-flux), and so on.
+
+### karmada-aggregated-apiserver
+
+This component is used to access resources on the member cluster through karmada-apiserver,
+achieving unified authentication on the Karmada control plane and authentication through user impersonation.
+
+The resources installed in the aggregate apiserver include clusters, clusters/status, clusters/proxy.
+
+### kube-controller-manager
+
+This component uses the official image of Kuberntes, the difference is that only several controllers are
+enabled such as namespace, garbagecollector, serviceaccount-token controllers.
```

Comment:
Can we directly attach the relevant instructions to ?

Reply:
done